public/app/plugins/datasource/azuremonitor/types/query.ts

```diff
@@ -0,0 +1,19 @@
+import { AzureMonitorQuery as AzureMonitorQueryBase, AzureQueryType } from '../dataquery.gen';
+
+export { AzureQueryType };
+export {
+  AzureMetricQuery,
+  AzureLogsQuery,
+  AzureResourceGraphQuery,
+  AzureMonitorResource,
+  AzureMetricDimension,
+  ResultFormat,
+} from '../dataquery.gen';
+
+/**
+ * Represents the query as it moves through the frontend query editor and datasource files.
+ * It can represent new queries that are still being edited, so all properties are optional
+ */
+export interface AzureMonitorQuery extends AzureMonitorQueryBase {
+  queryType?: AzureQueryType;
```

Comment:
Can we also document that it's a workaround until the type extensions in cue are fixed? A TODO comment would be great, so we remember to fix it later.

Reply:
Yes I'll make sure to do this!